src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,171 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+import { PrinterDropdown } from './printer_dropdown.js';
+
+export class PrinterManager {
+
+  #adultDropdown;
+  #minorDropdown;
+  #dropdowns;
+
+  #printerType = LP2844;
+
+  /**
+   * Initializes a new instance of the PrinterManager class.
+   *
+   * @param {Navigator} nav - Browser Navigator to hook up events to.
+   * @param {PrinterDropdown} adultDropdown - The dropdown object managing the adult printer.
+   * @param {PrinterDropdown} minorDropdown - The dropdown object managing the Minor printer.
+   */
+  constructor(nav, adultDropdown, minorDropdown) {
+    this.#adultDropdown = adultDropdown;
+    this.#adultDropdown.addEventListener(PrinterDropdown.ConnectEventName, async (e) => this.handleRequestConnect(e));
+    this.#adultDropdown.addEventListener(PrinterDropdown.DisconnectEventName, async (e) => this.handleRequestDisconnect(e));
+
+    this.#minorDropdown = minorDropdown;
+    this.#minorDropdown.addEventListener(PrinterDropdown.ConnectEventName, async (e) => this.handleRequestConnect(e));
+    this.#minorDropdown.addEventListener(PrinterDropdown.DisconnectEventName, async (e) => this.handleRequestDisconnect(e));
+
+    // Convenience array
+    this.#dropdowns = [this.#adultDropdown, this.#minorDropdown];
+
+    nav.usb.addEventListener('connect', async (e) => this.handleConnectPrinter(e));
+    nav.usb.addEventListener('disconnect', async (e) => this.handleDisconnectPrinter(e));
+  }
+
+  /**
+   * Assign a printer to a dropdown
+   * @param {LP2844} printer
+   * @param {PrinterDropdown} dropdown
+   */
+  assignPrinter(printer, dropdown) {
+    // Find the first dropdown in order, hope it's the right one!
+    // TODO: Make this smarter and pull from a cache or something for assignment
+    // instead of just first past the post.
+    dropdown = dropdown || this.#dropdowns.find(d => !d.printer);
+    if (dropdown) {
+      dropdown.setPrinter(printer);
+    } else {
+      console.warn("Connected to a printer that we couldn't assign anywhere!");
+    }
+  }
+
+  /**
+   * Handle a recquest to add a printer to a dropdown.
+   * @param {CustomEvent} o - PrinterDropdown connect event
+   * @param {PrinterDropdown} o.target - The dropdown target.
+   */
+  async handleRequestConnect({ target: dropdown }) {
+    if (dropdown.printer) {
+      // Spurious event dispatch?
+      return;
+    }
+
+    this.pairPrinter(dropdown);
+  }
+
+  /**
+   *
+   * @param {Object} o - Event Object.
+   * @param {USBDevice} o.device - The USB device to connect to as a printer.
```

Comment:
I don't think this is correct?

Reply:
Same as the other message, this is how you describe parameter destructuring.

Reply:
oh interesting, cool!